test/specs/core.interaction.tests.js

```diff
@@ -912,4 +912,62 @@ describe('Core.Interaction', function() {
       expect(elements).toContain(firstElement);
     });
   });
+
+  it('should select closest non-null elements if spanGaps=true and closest non-null element is to the left', function() {
+    const chart = window.acquireChart({
+      type: 'line',
+      data: {
+        labels: [1, 2, 3, 4, 5, 6, 7, 8, 9],
+        datasets: [{
+          data: [12, 19, null, null, null, null, 5, 2, 3],
+          spanGaps: true,
+        }]
+      }
+    });
+    chart.update();
+    const interactionPointIndex = 3;
+    const meta = chart.getDatasetMeta(0);
+    const point = meta.data[interactionPointIndex];
+
+    const evt = {
+      type: 'click',
+      chart: chart,
+      native: true, // needed otherwise things its a DOM event
+      x: point.x,
+      y: point.y,
+    };
+
+    const expectedInteractionPointIndex = 1;
+    const elements = Chart.Interaction.modes.nearest(chart, evt, {axis: 'x', intersect: false}).map(item => item.element);
+    expect(elements).toEqual([meta.data[expectedInteractionPointIndex]]);
+  });
+
+  it('should select closest non-null elements if spanGaps=true and closest non-null element is to the right', function() {
+    const chart = window.acquireChart({
+      type: 'line',
+      data: {
+        labels: [1, 2, 3, 4, 5, 6, 7, 8, 9],
+        datasets: [{
+          data: [12, 19, null, null, null, null, 5, 2, 3],
```

Comment:
thought: maybe it would be good to have test(s) for the zero and edge cases, something like: 

```suggestion
  it.each([
    [12, 19, null, null, null, null, 5, 2, 3],
    [12, 19, null, null, null, null, 0, 2, 3],
    [12, 19, null, null, null, null, -1],]
  ])('should select closest non-null elements if spanGaps=true and closest non-null element is to the right', function(data) {
    const chart = window.acquireChart({
      type: 'line',
      data: {
        labels: [1, 2, 3, 4, 5, 6, 7, 8, 9],
        datasets: [{
          data,
```

Reply:
Makes sense. I have now replace the two tests I had to one test with 12 test cases. The syntax is slightly different than the one you proposed, because `it.each` is not possible in Jasmine. I can see this syntax is available in Jest, but I cannot see it being used in this repository.